product_template_pricelist_button/models/product_template.py

```diff
@@ -0,0 +1,38 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import models
+from odoo.osv import expression
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    def open_pricelist_rules(self):
+        action = super().open_pricelist_rules()
+        domain = [
+            '|',
+            ('product_tmpl_id', '=', self.id),
+            ('product_id', 'in', self.product_variant_ids.ids),
+            ]
```

Comment:
The purpose of adding the new fragment and reusing existing one. This fragment of code is, again, overwritting code. We should retrieve current domain from obtained `action` object.